File: examples/swift/constructs/functions_recursion.swift
// Ported from examples/lumen/constructs/functions_recursion.lm by scripts/port_examples.py; edit the Lumen original, not this file.
func factorial(n: Int) -> Int {
    if n <= 1 {
        return 1
    } else {
        return n * factorial(n: n - 1)
    }
}

func countdown(n: Int) -> Int {
    if n <= 0 {
        print("Done")
    } else {
        print(n)
        return countdown(n: n - 1)
    }
}

print("Test: Recursion")
print("Factorial of 5:")
print(factorial(n: 5))
print("Countdown from 3:")
countdown(n: 3)

File: examples/javascript/constructs/functions_recursion.js
// Ported from examples/lumen/constructs/functions_recursion.lm by scripts/port_examples.py; edit the Lumen original, not this file.
function factorial(n) {
    if (n <= 1) {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}

function countdown(n) {
    if (n <= 0) {
        console.log("Done");
    } else {
        console.log(n);
        return countdown(n - 1);
    }
}

console.log("Test: Recursion");
console.log("Factorial of 5:");
console.log(factorial(5));
console.log("Countdown from 3:");
countdown(3);
